src/Items/ItemBoat.h

```diff
@@ -98,6 +98,12 @@ class cItemBoatHandler :
 		cBoat * Boat = new cBoat(x + 0.5, y + 0.5, z + 0.5);
 		Boat->Initialize(*a_World);
 
+		// Remove boat from players hand
+		if (!a_Player->IsGameModeCreative())
```

Comment:
What if the player is in Spectator mode? Should they be allowed to place boats, and what to do then?